lib/parser/javascript.js

```diff
@@ -199,9 +179,6 @@ ReplyParser.prototype.execute = function (buffer) {
         } catch (err) {
             // catch the error (not enough data), rewind, and wait
             // for the next packet to appear
-            if (! (err instanceof IncompleteReadBuffer)) {
```

Comment:
This is dead code after fixing the issue that user errors got catched by the parser.